RubberduckTests/Inspections/UnreachableCaseInspectionTests.cs

```diff
@@ -0,0 +1,2398 @@
+        public void UciUnit_LogicUnaryConstants(string operands, string expected)
+        {
+            GetUnaryOpValues(operands, out IParseTreeValue theValue, out string opSymbol);
+
+            var result = Calculator.Evaluate(theValue, opSymbol);
+
+            Assert.AreEqual(expected, result.ValueText);
+            Assert.IsTrue(result.ParsesToConstantValue, "Expected IsConstantValue field to be 'True'");
+        }
+
+        [TestCase("-_45", "-45")]
+        [TestCase("-_23.78", "-23.78")]
+        [TestCase("-_True", "True?Boolean")]
+        [TestCase("-_False", "False?Boolean")]
+        [TestCase("-_1?Double", "-1?Double")]
+        [TestCase("-_-1?Double", "1?Double")]
+        [Category("Inspections")]
+        public void UciUnit_MinusUnaryOp(string operands, string expected)
+        {
+            var expectedVal = CreateInspValueFrom(expected);
+            GetUnaryOpValues(operands, out IParseTreeValue LHS, out string opSymbol);
+            var result = Calculator.Evaluate(LHS, opSymbol);
+
+            Assert.AreEqual(expectedVal.ValueText, result.ValueText);
+            Assert.IsTrue(result.ParsesToConstantValue);
+        }
+
+        [TestCase("IsLT=5", "", "IsLT=5")]
+        [TestCase("IsGT=5", "", "IsGT=5")]
+        [TestCase("IsLT=5", "IsGT=300", "IsLT=5!IsGT=300")]
+        [TestCase("IsLT=5,Range=45:55", "IsGT=300", "IsLT=5!IsGT=300!Range=45:55")]
+        [TestCase("IsLT=5,Range=45:55", "IsGT=300,Single=200", "IsLT=5!IsGT=300!Range=45:55!Single=200")]
+        [TestCase("IsLT=-2,Range=45:55", "IsGT=300,Single=200,RelOp=x < 50", "IsLT=-2!IsGT=300!Range=45:55!Single=200!RelOp=x < 50")]
+        [TestCase("Range=45:55", "Range=60:65", "Range=45:55,60:65")]
+        [TestCase("Single=45,Single=46", "Single=60", "Single=45,46,60")]
+        [TestCase("RelOp=x < 50", "RelOp=x > 75", "RelOp=x < 50,x > 75")]
+        [Category("Inspections")]
+        public void UciUnit_ToString(string firstCase, string secondCase, string expected)
+        {
+            var filters = RangeDescriptorsToFilters(new string[] { firstCase, secondCase/*, expectedClauses*/ }, Tokens.Long);
+            filters[0].Add(filters[1]);
+
+            Assert.AreEqual(expected, filters[0].ToString());
+        }
+
+        [TestCase("50?Long_To_x?Long", "Long", "Range=50:x")]
+        [TestCase("50?Long_To_100?Long", "Long", "Range=50:100")]
+        [TestCase("Soup?String_To_Nuts?String", "String", "Range=Nuts:Soup")]
+        [TestCase("50.3?Double_To_100.2?Double", "Long", "Range=50:100")]
+        [TestCase("50.3?Double_To_100.2?Double", "Double", "Range=50.3:100.2")]
+        [TestCase("50_To_100,75_To_125", "Long", "Range=50:100,Range=75:125")]
+        [TestCase("50_To_100,175_To_225", "Long", "Range=50:100,Range=175:225")]
+        [TestCase("500?Long_To_100?Long", "Long", "Range=100:500")]
+        [Category("Inspections")]
+        public void UciUnit_AddRangeClauses(string firstCase, string filterTypeName, string expectedClauses)
```

Comment:
I think we need test case for silly not-a-range ranges like `Case 1 To 1` which is legal.

Reply:
Done